core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
I would actually prefer to not have `apply` and just have anyone who wants the behavior to call `.get.get`, as @jedws said. I would much prefer to see only safe FP functions exist in Cats.

Afaik we don't have instances for `Seq` - `Try` is a bit sketchy. I believe the argument was if you only use pure functions, then `Try` is lawful, it basically becomes a glorified `Either[Throwable, A]`.